packages/react-native-renderer/src/ReactFabricRenderer.js

```diff
@@ -203,11 +225,30 @@ const ReactFabricRenderer = ReactFiberReconciler({
     return false;
   },
 
-  getRootHostContext(): {} {
-    return emptyObject;
+  getRootHostContext(rootContainerInstance: Container): HostContext {
+    return __DEV__ ? {isInAParentText: false} : emptyObject;
```

Comment:
I'm surprised that `emptyObject` fulfills the type definition for `HostContext`. I would've expected `isInAParentText` to need to be optional (`isInAParentText?: boolean`), but maybe `emptyObject` is typed with something lossy like `Object`?

Reply:
That's true. I'd guess `emptyObject` is resolving to an `any` type?

Reply:
Actually something more insidious is going on here. I can't make Flow report an error in this type no matter what I do, within the React native or fabric renderers.

Reply:
For example, given this type:
```js
type HostContext = $ReadOnly<{|
  isInAParentText: boolean,
|}>;
```

Flow reports "no errors" if I add these lines inside of a method like `createInstance()` that accepts a `hostContext: HostContext` param:
```js
hostContext.isInAParentText = false;
```

But if I change it to this, Flow fails:
```js
(hostContext: HostContext).isInAParentText = false;
```

> Covariant property `isInAParentText` incompatible with contravariant use in assignment of property `isInAParentText`

Reply:
Note that the React repo is using `flow-cli` v0.61 and the latest version is v0.72 ~ not sure if this could be an old Flow bug.

Reply:
Resolution: I was trying to reproduce inside of a conditional, and we're using an older version of Flow that has a bug (reproducible in the REPL) that does not properly handle this case.

Reply:
Yes. I'll add this discussion point to our sync notes.

Reply:
Here is a failing case:

```js
type HostContext = $ReadOnly<{|
  isInAParentText: boolean                  
|}>;

const hostContext: HostContext = {
  isInAParentText: true
};

if (hostContext.thisPropertyDoesNotExist) {
  // And this one is a Boolean and should not be writable
  hostContext.isInAParentText = 'abc';
}
```

* [Flow 0.61.0 link](https://flow.org/try/#0C4TwDgpgBAEg9gZ2AYTgO2BAHsKBeKAEgCUIBDAEwHk0AbEAHgG8AfAKCigEsEBJNAIIAFMgCcIGACrZgALigAjOHFrk0nDZq2a2LAL4A+ANxs2AY3RIoAC0Qp0mHPPhJUGGfihMO3PoJHiUjLywKIArhBseiZsXABmUAAUtq4OMgB0wNY8QqJwkKKgACJwEAgAcnDAAKJYPMAAlF4+APQtUAJoFFBZPFDo0H1kUABCyqpk6pPdCLZhtN1oVYrQAO6iXMBkCqo+KfbuOOk8-MJiEsDSOJ4A5NtmNyZ6QA)
* [Flow 0.72.0 link](https://flow.org/try/#0C4TwDgpgBAEg9gZ2AYTgO2BAHsKBeKAEgCUIBDAEwHk0AbEAHgG8AfAKCigEsEBJNAIIAFMgCcIGACrZgALigAjOHFrk0nDZq2a2LAL4A+ANxs2AY3RIoAC0Qp0mHPPhJUGGfihMO3PoJHiUjLywKIArhBseiZsXABmUAAUtq4OMgB0wNY8QqJwkKKgACJwEAgAcnDAAKJYPMAAlF4+APQtUAJoFFBZPFDo0H1kUABCyqpk6pPdCLZhtN1oVYrQAO6iXMBkCqo+KfbuOOk8-MJiEsDSOJ4A5NtmNyZ6QA)